docs/setup-guide/build_k8s_from_source/README.md

```diff
@@ -0,0 +1,145 @@
+- Click 'Add rule' for Inbound rules, following the below list
+```
+All TCP             TCP     0-65535     0.0.0.0/0
+All UDP             UDP     0-65535     0.0.0.0/0
+SSH                 TCP     22          0.0.0.0/0
+All ICMP - IPv4     ICMP    All         0.0.0.0/0
+```
+- Click 'Review and Launch'
+- Click 'Launch'
+- Select a key pair
+- Wait until the vm is running
+
+# Install K8s components and tools
+- SSH into the vm on AWS
+
+## Configure iptables
+- Run `sudo modprobe br_netfilter`
+- Run `lsmod | grep br_netfilter` to make sure br_netfilter module is loaded
+- Run
+  ```
+  cat <<EOF | sudo tee /etc/modules-load.d/k8s.conf
+  br_netfilter
+  EOF
+  ```
+- Run
+  ```
+  cat <<EOF | sudo tee /etc/sysctl.d/k8s.conf
+  net.bridge.bridge-nf-call-ip6tables = 1
+  net.bridge.bridge-nf-call-iptables = 1
+  EOF
+  ```
+- Run `sudo sysctl --system`
+
+## Install runtime docker
+- Run `sudo apt-get -y update; sudo apt-get -y install docker.io`
+- Run `sudo systemctl enable docker.service`
+- Run `sudo groupadd docker`
+- Run `sudo usermod -aG docker $USER`
+- Reboot the machine by running `sudo reboot`
+- Verify you can run `docker` without `sudo` by running `docker run hello-world`
+- Run `sudo mkdir /etc/docker`
+- Run 
+```
+cat <<EOF | sudo tee /etc/docker/daemon.json
+{
+  "exec-opts": ["native.cgroupdriver=systemd"],
+  "log-driver": "json-file",
+  "log-opts": {
+    "max-size": "100m"
+  },
+  "storage-driver": "overlay2"
+}
+EOF
+```
+- Run `sudo systemctl daemon-reload`
+- Run `sudo systemctl restart docker`
+
+## Install GoLang
+- Run `cd ~`
+- Run `sudo apt-get -y install make gcc jq`
+- Run `wget https://redirector.gvt1.com/edgedl/go/go1.17.1.linux-amd64.tar.gz -O go1.17.1.tar.gz`
+- Run `tar -xzvf go1.17.1.tar.gz`
+- Run `mkdir gopath`
+- Append the following lines to `~/.bashrc`
+  ```
+  export GOROOT=$HOME/go
+  export GOPATH=$HOME/gopath
+  export PATH=$GOROOT/bin:$PATH
+  ```
+- Run `source ~/.bashrc`
+- Check `go version`
+
+## Build Kubernetes from source
+- Run `sudo apt -y install conntrack socat`
+- Run `mkdir -p $GOPATH/src/k8s.io`
+- Run `cd $GOPATH/src/k8s.io`
+- Run `git clone https://github.com/kubernetes/kubernetes.git`
+- Run `cd kubernetes`
+- Run `git fetch --all --tags`
+- Run `git checkout tags/v1.21.0`
+- Run `make clean`
+- Run `make all`
+- Install CNI plugins by running:
+```
+CNI_VERSION="v0.8.2"
+ARCH="amd64"
+sudo mkdir -p /opt/cni/bin
+curl -L "https://github.com/containernetworking/plugins/releases/download/${CNI_VERSION}/cni-plugins-linux-${ARCH}-${CNI_VERSION}.tgz" | sudo tar -C /opt/cni/bin -xz
+```
+- Install crictl by running:
+```
+DOWNLOAD_DIR=/usr/local/bin
+sudo mkdir -p $DOWNLOAD_DIR
+CRICTL_VERSION="v1.22.0"
+ARCH="amd64"
+curl -L "https://github.com/kubernetes-sigs/cri-tools/releases/download/${CRICTL_VERSION}/crictl-${CRICTL_VERSION}-linux-${ARCH}.tar.gz" | sudo tar -C $DOWNLOAD_DIR -xz
+```
+- Run `sudo ln -s $GOPATH/src/k8s.io/kubernetes/_output/bin/kubelet /usr/bin/`
```

Comment:
stretch goal, use a loop here to prevent repetition. not required.

see [here](https://github.com/pdgetrf/mizar_cluster_scripts/blob/8dfc0c2a28a6be2c61eaa0c94a668f464d5aa7c1/build_docker_image.sh#L25) as an example

Reply:
added a loop in both manual guide and script (setup_3.sh)